animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -140,37 +150,51 @@ cdef class Mesh:
         if self.getAttributeDomainLength(attribute.domain) != len(attribute.data):
             raise Exception("invalid length")
 
-        self.attributes[attribute.name] = attribute
+        if attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        elif attribute.type == AttributeType.UV_MAP:
+            self.uvMapAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.VERTEX_COLOR:
+            self.vertexColorAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        else:
+            self.customAttributes[attribute.name] = attribute
 
     def getAttributes(self, AttributeType type):
-        attributes = list()
-        for name, attribute in self.attributes.items():
-            if type == attribute.type: attributes.append((name, attribute))
-            elif type == AttributeType.VERTEX_COLOR and attribute.dataType == AttributeDataType.BYTE_COLOR:
-                attributes.append((name, attribute))
-        return attributes
+        if type == AttributeType.UV_MAP:
+            return list(self.uvMapAttributes.items())
+        elif type == AttributeType.VERTEX_COLOR:
+            return list(self.vertexColorAttributes.items())
+        else:
+            attributes = list()
+            for name, attribute in self.customAttributes.items():
+                if type == attribute.type: attributes.append((name, attribute))
+            return attributes
```

Comment:
Why the special handing here. Won't `type` always be equal to `attribute.type`?

Reply:
I have a query, are we going to separate other `types` of attributes as well *e.g.,* Vertex Weight, Bevel Vertex/Edge Weight, etc.? If so, then we do not need this special handling?

Reply:
@3DSinghVFX That's a good point. I didn't though about future additions. Give me second to think about this.

Reply:
@3DSinghVFX What do you think about creating a so called `builtInAttributes`. That dictionary should hold single things  like Material Indices, Bevel Vertex/Edge weights, and so on.

Reply:
Other possible attributes are: 
1. Vertex Weight
2. Bevel Vertex Weight
3. Bevel Edge Weight
4. Crease
5. Seams
6. Face Groups

Reply:
@3DSinghVFX  Vertex Weights and Face Maps are not really built-in, they are more similar to vertex colors and uv maps.

An ordered dictionary `builtInAttributes` can hold all of the single built-in attributes like your list [2, 5]. Rename the `getMeshAttributes` method to `getAttributeDictionaries`. Instead of returning a list of tuples, just return a list of dictionaries, the second element is never user in the tuple. Then adjust your code accordingly. Some of my review items will no longer stand, but finish up and I will rereview later.

Reply:
Okay.

Reply:
Another thing I would like to mention here is that *Vertex Color* is shared in *Custom Attribute*. So, in the current patch, I have added a condition in Vertex Color and Custom Attribute Nodes to avoid the same name conflict.

Should we handle that in the *InsertAttribute* method in `mesh_data.pyx` file?

Reply:
@3DSinghVFX hmmm, at this point I would say lets separate them completely. We can divert from Blender in this case. Custom attribute nodes should only deal with custom attributes no other data. We already having uv and vertex color nodes to deal with other things.

Reply:
Omar, they are separated but we need to check only the name of Vertex Color and Custom Attribute (Byte Color). Otherwise, Vertex Color and Custom Attribute (Byte Color) with the same name will override each other in the node-tree,
![Screenshot from 2021-04-24 09-06-45](https://user-images.githubusercontent.com/30294746/115951435-533bab80-a4fe-11eb-9e5e-1adbbdf07131.png)

Reply:
@3DSinghVFX  Lets leave it at that even with the override. The last thing I want is for the Get Custom Attribute node to only return custom attributes, we are diverting from the Blender logic to make things more intuitive.

Reply:
Okay.